### PR TITLE
Improve page meta data to help URL unfurling

### PR DESCRIPTION
**Pull request by teneightfive**
This change improves the meta data that is pulled in from other sources when a content item is shared.

Currently we don't provide any meta descriptions. This change also uses the short copy from the local header for page descriptions to give a better insight into the content of the page.

## How it looks

Once deployed the URL will reflect the production URL.

### Facebook

screen shot 2017-01-19 at 17 23 36

### Twitter

screen shot 2017-01-19 at 17 23 01

### Slack

screen shot 2017-01-19 at 17 24 11

## Future

Slack and Twitter also support the addition of label:data pairs to provide extra information. Some examples can include the price of an item or its availability. It might be good to investigate if including supplementary data might help meet user needs for certain conditions. 

Possible things to include could be symptoms of a condition or when to see a GP/go to A&E.

One of the developers at Slack has written a [blog post](https://medium.com/slack-developer-blog/everything-you-ever-wanted-to-know-about-unfurling-but-were-afraid-to-ask-or-how-to-make-your-e64b4bb9254#.n8shdmam5) about improving the _unfurling_ of pages. It includes some examples of good unfurls.

### CODE REVIEW

File: app/middleware/locals.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ module.exports = (config) => {
     res.locals.CDN_HOST = config.staticCdn;
     res.locals.FONT_CDN_HOST = config.fontCdn;
     res.locals.ORIGINAL_URL = req.originalUrl;
+    res.locals.FULL_URL = `${(req.isHttps ? 'https' : req.protocol)}://${req.get('host')}${req.originalUrl}`;
```
Comment (tyom) [inline app/middleware/locals.js]: `CANONICAL_URL`?

File: app/middleware/locals.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ module.exports = (config) => {
     res.locals.CDN_HOST = config.staticCdn;
     res.locals.FONT_CDN_HOST = config.fontCdn;
     res.locals.ORIGINAL_URL = req.originalUrl;
```
Comment (tyom) [inline app/middleware/locals.js]: Is this a URL or a path?

Comment (teneightfive) [inline app/middleware/locals.js]: [Express docs](https://expressjs.com/en/api.html#req.originalUrl)

>In a middleware function, req.originalUrl is a combination of req.baseUrl and req.path

**Comment by tyom** [comment]
Looks like `description` is almost always the same as the summary in the local header. Do you think we can reduce that duplication somehow?

File: app/views/_includes/global/head.nunjucks [hunk 3]
```diff
@@ -2,12 +2,8 @@
 <meta name="viewport" content="width=device-width, initial-scale=1">
 <meta http-equiv="x-ua-compatible" content="ie=edge">
 
-{% if CDN_HOST != '/' %}
-  <link href="{{ CDN_HOST }}" rel="dns-prefetch">
-{% endif %}
-{% if FONT_CDN_HOST != '/' %}
-  <link href="{{ FONT_CDN_HOST }}" rel="dns-prefetch">
-{% endif %}
+{% if CDN_HOST %}<link href="{{ CDN_HOST }}" rel="dns-prefetch">{% endif %}
```
Comment (tyom) [inline app/views/_includes/global/head.nunjucks]: Why did you collapse these? I think it made it harder to scan.

**Comment by teneightfive** [comment]
@tyom I agree that duplication is not ideal in this situation but the rendering app should be kept simple. It should just expect a structured response and if fields like `description` exist it should render that.

We shouldn't try and do something in this app that looks within the header stream to see if a markdown component exists and use that content for description if it doesn't have a `description` key. 

If we decide to keep the short summary in the blue header we should look at having the publishing tool support sharing the page description for this component so that duplication is avoided in the application where the content is written, rather than in the application that is only concerned with rendering data to templates.

**Pull request merged**